Done (not tested)

PedroTrujilloV opened the pull request:
The requirements are satisfied (just need return to the map from the confirmation alert), but running the app shows :2016-10-19
00:36:33.508 Street Trees[41864:1657473] Can't find keyplane that
supports type 5 for keyboard iPhone-Portrait-PhonePad; using
3561727473_Portrait_iPhone-Complex-Pad_Default
fatal error: An Address has not been passed.: file
/Users/Pedro/energy-saving-trees-ios/Street
Trees/STUserDetailsFormViewController.swift, line 160

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

PedroTrujilloV committed:
test avoid error

error: Your local changes to the following files would be overwritten
by merge:
	Street Trees.xcodeproj/project.pbxproj
Please, commit your changes or stash them before you can merge.
Aborting
 (128)

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

PedroTrujilloV committed:
Merge remote-tracking branch 'cforlando/master'

PedroTrujilloV committed:
Confirmation Page view controller and story board for this.

PedroTrujilloV committed:
STConfirmationPageViewController:
- Table view methods
- Get information from Wufoo form

PedroTrujilloV committed:
STConfirmationPageViewController

Story board View Controller and Class

PedroTrujilloV committed:
Cell registration

cell registration for each field in the form

PedroTrujilloV committed:
fixed the mistake

a miss understand about the Wufoo form and the three objects.

PedroTrujilloV committed:
Protocol ready

- Confirmation Page
- Protocol in confirmation page with function at Order Form Navigation
controller setting the var wufooForm

PedroTrujilloV committed:
Done (no tested)

There requirements are done, but running the app shows :2016-10-19
00:36:33.508 Street Trees[41864:1657473] Can't find keyplane that
supports type 5 for keyboard iPhone-Portrait-PhonePad; using
3561727473_Portrait_iPhone-Complex-Pad_Default
fatal error: An Address has not been passed.: file
/Users/Pedro/energy-saving-trees-ios/Street
Trees/STUserDetailsFormViewController.swift, line 160

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+// MARK: - Protocol
+
+protocol STConfirmationPageViewControllerDelegate: NSObjectProtocol {
+    func ConfirmationFormViewController(form: STConfirmationPageViewController, didCompleteWithWufooForm anWufooForm: STTKWufooForm)
```
Change to: 
func confirmationFormViewController(form: STConfirmationPageViewController, didCompleteWithWufooForm aWufooForm: STTKWufooForm)

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+
+//**********************************************************************************************************************
+
+class STConfirmationPageViewController: UIViewController, UITableViewDataSource, UITableViewDelegate
```
Add the Address, Contact, and TreeDescription protocols

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+    var aContact: STTKContact?
+    
+    //just initial test values for vars
+    var treeName = "Example Tree"
```
Remove test values

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
```
remove commented code

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+        
+    }
+
+    override func didReceiveMemoryWarning()
```
remove unused function

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+        
+        //treeName = (aTree?.name)!
+        //treeImage =  aTree.
+        address = (anAddress?.streetAddress)!  + " " +  (anAddress?.secondaryAddress)!
```
remove test data configuration

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int
+    {
+        return 1
```
Create an enum for the cells and then create a datasource property or function that returns a data source struct that contains your content for the tableview. You should be able to query your new struct for the amount of sections and query for how many items are in a section.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
+    {
+        return 8
```
Use the datasource struct return the correct amount of rows for the section.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell
+    {
+        let cell = tableView.dequeueReusableCellWithIdentifier("emptyCell", forIndexPath: indexPath)
```
create a single cell in the tableview on the storyboard. looking at your code, a basic tableviewcell would be fine. Then you can query your datasource struct for the correct cell information.

I would ask the datasource for the enum at section and row and then create a switch for each enum variable instead of creating a ton of if statements. a large list of if statements can be rather cpu intensive and we don't want the cpu to do any additional work than what it has too.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+        alertConfimationMessage.addAction(sentAction)
+        
+        self.presentViewController(alertConfimationMessage, animated: true, completion: nil)
+        self.delegate?.ConfirmationFormViewController(self, didCompleteWithWufooForm: STTKWufooForm(tree: treeName, forContact: aContact!))
```
never force unwrap anything. create and if let or guard let to unwrap the contact object.

LondonAtlas commented on Street Trees/STOrderFormNavigationViewController.swift:
```diff
@@ -64,6 +66,15 @@ STAddressFormViewControllerDelegate {
+        
+        if let confirmationViewController = viewController as? STConfirmationPageViewController {
+            
+            confirmationViewController.treeName = (self.treeDescription?.name)!
```
not sure why this is wrapped in ( ). Also never for unwrap an object.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STOrderFormNavigationViewController.swift:
```diff
@@ -64,6 +66,15 @@ STAddressFormViewControllerDelegate {
             treeViewController.delegate = self
         }
+        
+        if let confirmationViewController = viewController as? STConfirmationPageViewController {
```
only assign the delegate in this if let. By conforming the confirmationViewController to the TreeDescription, Address, and Contact protocols it will be given the correct objects when it is placed on the navigation stack.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+    weak var  delegate:STConfirmationPageViewControllerDelegate?
+    
+
+    override func viewDidLoad()
```
remove this function as the table view will should use a datasource to insert its content.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees.xcodeproj/project.pbxproj:
```diff
@@ -1438,6 +1449,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = "com.CodeForOrlando.Street-Trees";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				PROVISIONING_PROFILE = "";
+				SWIFT_VERSION = 2.3;
```
Revert this to use swift 2.2. We do not support Xcode 8 or Swift 2.3 at this time.

PedroTrujilloV replied:
Done

LondonAtlas reviewed (changes requested):
This is a good starting point.

LondonAtlas commented on Street Trees/STConfirmationPageViewController.swift:
```diff
@@ -0,0 +1,168 @@
+        let alertConfimationMessage = UIAlertController(title: "Confirmation", message: "Thank You! Your request has been sent", preferredStyle: .Alert)
+        let sentImage = UIImage(named: "sent")
+        let sentAction = UIAlertAction(title: "Ok", style: .Default, handler: nil)
+        sentAction.setValue(sentImage, forKey: "image")
```
Don't force images into alerts as the look like the attached screenshot.
<img width="487" alt="screen shot 2016-10-20 at 9 08 53 am" src="https://cloud.githubusercontent.com/assets/1647012/19561031/043f115a-96a5-11e6-9cd2-9754d33c03bf.png">

LondonAtlas commented on Street Trees/ConfirmationPage.storyboard:
```diff
@@ -0,0 +1,230 @@
+                        <rect key="frame" x="0.0" y="0.0" width="600" height="600"/>
+                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+                        <subviews>
+                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Review" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="gGf-ik-9HV">
```
This review label is not required as the text in the navigation bar will be fine.

PedroTrujilloV replied:
Done

LondonAtlas commented on Street Trees/ConfirmationPage.storyboard:
```diff
@@ -0,0 +1,230 @@
+                                    <outlet property="delegate" destination="Qhd-d6-vZT" id="bYM-V8-8Qh"/>
+                                </connections>
+                            </tableView>
+                            <button opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="center" contentVerticalAlignment="center" lineBreakMode="middleTruncation" translatesAutoresizingMaskIntoConstraints="NO" id="8Bn-C0-Wfa">
```
remove this button and add a bar item to the right hand side of the navigation bar to keep it inline with the rest of the form. This is a deviation from the designs.

PedroTrujilloV replied:
Done

PedroTrujilloV committed:
Done:
- func confirmationFormViewController(form: STConfirmationPageViewController, didCompleteWithWufooForm aWufooForm: STTKWufooForm
- Remove test values
- remove this function as the table view will should use a datasource to insert its content.
- remove test data configuration
- remove unused function
- remove commented code
- Create an enum for the cells and then create a datasource property or function that returns a data source struct that contains your content for the tableview. You should be able to query your new struct for the amount of sections and query for how many items are in a section.
- Use the datasource struct return the correct amount of rows for the section.
- create a single cell in the tableview on the storyboard. looking at your code, a basic tableviewcell would be fine. Then you can query your datasource struct for the correct cell information.

I would ask the datasource for the enum at section and row and then create a switch for each enum variable instead of creating a ton of if statements. a large list of if statements can be rather cpu intensive and we don't want the cpu to do any additional work than what it has too.

-  only assign the delegate in this if let. By conforming the confirmationViewController to the TreeDescription, Address, and Contact protocols it will be given the correct objects when it is placed on the navigation stack.
- Revert this to use swift 2.2. We do not support Xcode 8 or Swift 2.3 at this time.
- This review label is not required as the text in the navigation bar will be fine.

PedroTrujilloV committed:
Done
-remove this button and add a bar item to the right hand side of the navigation bar to keep it inline with the rest of the form. This is a deviation from the designs.

PedroTrujilloV committed:
add image pdf to alert controller

PedroTrujilloV committed:
PR Commit
